app/src/components/common/customizable_tokens_select/modal_collateral.tsx

```diff
@@ -1,137 +1,107 @@
     : null
 
-  const onClickCloseButton = () => {
-    onClose()
-  }
-
   const onClickSaveButton = () => {
-    if (collateralWithAddress) {
-      onSave(collateralWithAddress)
+    if (validCollateralAddress) {
+      onSave(validCollateralAddress)
       onClose()
     }
   }
 
-  const portal: any = document.getElementById('portalContainer')
-
   const tokenDetails = () => {
-    return (
+    return validCollateralAddress ? (
       <>
-        <SubsectionTitle>Details</SubsectionTitle>
+        <SubsectionTitleStyled>Details</SubsectionTitleStyled>
         <Grid>
-          <TitleValue
+          <TitleValueStyled
             title={'Symbol:'}
-            value={collateralWithAddress && collateralWithAddress.symbol}
+            value={validCollateralAddress && validCollateralAddress.symbol}
           />
-          <TitleValue
+          <TitleValueStyled
             title={'Decimals:'}
-            value={collateralWithAddress && collateralWithAddress.decimals}
+            value={validCollateralAddress && validCollateralAddress.decimals}
           />
         </Grid>
       </>
-    )
+    ) : null
   }
 
-  const messageToRender = (
-    <Wrapper>
-      <Card>
-        <SubsectionTitle>Add a valid collateral token address</SubsectionTitle>
-        <CollateralAddressWrapper
-          formField={
-            <Textfield
-              name="collateralAddress"
-              onChange={(event: ChangeEvent<HTMLInputElement>) => {
-                const { value } = event.target
-                setCollateralAddress(value)
-              }}
-              placeholder="Type in a collateral address..."
-              type="text"
-            />
-          }
-          title={'Collateral token address'}
-          note={<ErrorStyled>{errorMessage || ''}</ErrorStyled>}
-        />
-        {tokenDetails()}
-        <ButtonContainer>
-          <ButtonStyled disabled={!collateralWithAddress} onClick={onClickSaveButton}>
-            Save
-          </ButtonStyled>
-          <Button buttonType={ButtonType.secondary} onClick={onClickCloseButton}>
-            Close
-          </Button>
-        </ButtonContainer>
-      </Card>
-    </Wrapper>
+  return (
+    <ModalWrapper isOpen={isOpen} onRequestClose={onClose} title={`Add Collateral`}>
+      <FormRow
+        formField={
+          <Textfield
+            hasError={errorMessage ? true : false}
+            hasSuccess={validCollateralAddress ? true : false}
+            name="collateralAddress"
+            onChange={(event: ChangeEvent<HTMLInputElement>) => {
+              const { value } = event.target
+              setCollateralAddress(value.trim())
+            }}
+            placeholder="0xac5d29e67a53ee4903d59a4c929b718e1d575eee"
+            type="text"
+          />
+        }
+        error={errorMessage || ''}
+        title={'Collateral Token Address'}
+        tooltip={{ id: 'ERC20', description: 'Enter a valid ERC20 address.' }}
```

Comment:
Look really great! I have an small change, I think the text is not the same as in the designs, Can we update it?

Reply:
I changed it on purpose, I like my version more. I find the original a little redundant.